Userland/Libraries/LibWeb/CSS/StyleResolver.cpp

```diff
@@ -765,8 +765,11 @@ static void set_property_expanding_shorthands(StyleProperties& style, CSS::Prope
     style.set_property(property_id, value);
 }
 
-StyleResolver::CustomPropertyResolutionTuple StyleResolver::resolve_custom_property_with_specificity(const DOM::Element& element, const String& custom_property_name) const
+StyleResolver::CustomPropertyResolutionTuple StyleResolver::resolve_custom_property_with_specificity(DOM::Element& element, const String& custom_property_name) const
 {
+    if (element.has_custom_property(custom_property_name))
+        return element.resolve_custom_property(custom_property_name).value();
```

Comment:
what if this fails / returns an empty optional?

Reply:
Ah, it can't fail because you're checking it.
how about checking only once?
```c++
if (auto maybe_property = element.resolve_custom_property(custom_property_name); maybe_property.has_value())
    return maybe_property.value();
```

Reply:
Good point! This also eliminates the need for has_custom_property().